src/plugins/formulas/formulas.js

```diff
@@ -203,36 +250,233 @@ export class Formulas extends BasePlugin {
       });
     }
 
+    const hfSetUpProperly = this.setupHF();
+
+    if (!hfSetUpProperly) {
+      return;
+    }
+
+    // HyperFormula events:
+    this.hyperformula.on('valuesUpdated', (...args) => this.onHFvaluesUpdated(...args));
+
+    this.applyHFSettings();
+
     super.enablePlugin();
   }
 
   /**
    * Disables the plugin functionality for this Handsontable instance.
    */
   disablePlugin() {
-    // TODO add tests for this line
-    this.hyperformula.destroy();
-
     super.disablePlugin();
   }
 
+  /**
+   * Triggered on `updateSettings`.
+   */
+  updatePlugin() {
+    this.#settings = this.hot.getSettings()[PLUGIN_KEY];
+
+    this.applyHFSettings();
```

Comment:
Shouldn't `this.#settings = this.hot.getSettings()[PLUGIN_KEY];` be inside `applyHFSettings`? or passed as an argument. `applyHFSettings` is now dependent whether someone set the `#settings` or not. Not very "pure" and prone to side effects

Reply:
`this.#settings` is [defined initially when the plugin loads](https://github.com/handsontable/handsontable/blob/ffeeebe25e4d97af410d3c4642414576ccb12812/src/plugins/formulas/formulas.js#L35), so `applyHFSettings()` applies the settings, that are currently loaded by the plugin (which should not happen anywhere but init/updateSettings), so don't think it's _that_ much of a problem.

I could move this assignment to the inside of `applyHFSettings`, but I think that might not be as readable - currently, this line serves as a "plugin was updated -> update the settings cache" kind of thing. If it was inside of the apply function, it would work just as fine, though, so your call.

Reply:
Dismissed by the new review